### ISSUE TITLE
doc: document everything!

### ISSUE THREAD

**Issue by OlegHahm** [RIOT-OS/RIOT#2137]
~~While RIOT's core and most of newer code is comparably well documented the rest could need some improvement. Currently (after #2117 is merged) we have~~

~~\- [ ] 534 warnings in `boards`~~
~~\- [x] 2 warnings in `core` (see #2138)~~
~~\- [ ] 505 warnings in `cpu`~~
~~\- [ ] 432 warnings in `drivers` (see #2153)~~
~~\- [ ] 643 warnings in `sys/net`~~
~~\- [x] 126 warnings in the rest of `sys` (see #2149)~~

Let's focus on fixing these warnings.

_Updated numbers:_
- [ ] 23 files with warnings in `boards`
- [ ] 2 files with warnings in `core`
- [ ] 41 files with warnings in `cpu`
- [ ] 21 files with warnings in `drivers`
- [ ] 26 files with warnings in `sys/net`
- [ ] 14 files with warnings in the rest of `sys`

**Comment by OlegHahm** [comment]
Updated numbers:

- [ ] 23 warnings in `boards`
- [ ] 4 warnings in `core`
- [ ] 41 warnings in `cpu`
- [ ] 21 warnings in `drivers`
- [ ] 26 warnings in `sys/net`
- [ ] 14 warnings in the rest of `sys`

**Comment by OlegHahm** [comment]
#2837 fixes `core`.

**Comment by OlegHahm** [comment]
Retagged. Still some work to do until March.

**Comment by OlegHahm** [comment]
And another release postponed.

**Comment by kYc0o** [comment]
@OlegHahm can you give an update on this issue? I'm afraid it will be postponed again.

**Comment by OlegHahm** [comment]
```
make doc 2>&1 | grep -i warn | grep -oE 'RIOT\/[^\/]*\/' | cut -c 6- | sort | uniq -c
    432 boards/
      2 core/
    247 cpu/
      1 doc/
     62 drivers/
      3 pkg/
    170 sys/
```

**Comment by kYc0o** [comment]
wow... huge numbers indeed... well, let's try to do something for the next release.

**Comment by miri64** [comment]
Hope we can fix this for next release! :-)

**Comment by aabadie** [comment]
AFAICT, all warnings in documentation are fixed. Should we close this one ?

(tested with `make doc 2>&1 | grep -i warn | grep -oE 'RIOT\/[^\/]*\/' | cut -c 6- | sort | uniq -c`)

**Comment by miri64** [comment]
Yes.
